libkbfs/mdserver_remote.go

```diff
@@ -385,10 +388,30 @@ func (md *MDServerRemote) cancelObservers() {
 	}
 }
 
+// CancelRegistration implements the MDServer interface for MDServerRemote.
+func (md *MDServerRemote) CancelRegistration(ctx context.Context, id tlf.ID) {
+	md.observerMu.Lock()
+	defer md.observerMu.Unlock()
+	observerChan, ok := md.observers[id]
+	if !ok {
+		// not registered
+		return
+	}
+
+	// signal that we've seen the update
+	md.signalObserverLocked(
+		observerChan, id, errors.New("Registration canceled"))
+	// Setting nil here indicates that the server thinks we're still
+	// registered, though locally no one is listening.
+	md.observers[id] = nil
```

Comment:
Ah, yeah that's confusing.  The actual server in the backend still thinks we are registered, because we registered previously and never told the server to stop sending us updates (there is no RPC for that, besides killing our connection).  The comment is referring to the local `MDServerRemote` instance.  I'll fix up that interface comment to be more clear.